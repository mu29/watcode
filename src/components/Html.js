/* eslint-disable react/no-danger */
import React from 'react'
import { PropTypes } from 'prop-types'
import { Helmet } from 'react-helmet'

const Html = ({ styles, state, children }) => {
  const helmet = Helmet.renderStatic()
  const htmlAttrs = helmet.htmlAttributes.toComponent()
  const bodyAttrs = helmet.bodyAttributes.toComponent()

  return (
    <html lang="ko" { ...htmlAttrs }>
      <head>
        { helmet.title.toComponent() }
        { helmet.meta.toComponent() }
        { helmet.link.toComponent() }
        <meta charSet="utf8" />
        <link href="/style.css" rel="stylesheet" type="text/css" />
        <link href="//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css" rel="stylesheet" type="text/css" />
        <script defer src="https://use.fontawesome.com/releases/v5.0.13/js/all.js" integrity="sha384-xymdQtn1n3lH2wcu0qhcdaOpQwyoarkgLVxC/wZ5q7h9gHtxICrpcaSUfygqZGOe" crossOrigin="anonymous" />
        { styles }
      </head>
      <body { ...bodyAttrs }>
        <main id="app">{ children }</main>
        <script dangerouslySetInnerHTML={{ __html: state }} />
        <script src="/client.js" />
      </body>
    </html>
  )
}

Html.propTypes = {
  styles: PropTypes.node.isRequired,
  state: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default Html
