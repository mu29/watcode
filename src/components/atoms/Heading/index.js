import React from 'react'
import { PropTypes } from 'prop-types'
import styled, { css } from 'styled-components'
import {
  fontWeight,
  fontFamily,
  fontSize as getFontSize,
  color,
} from 'services/style'

export const calculateFontSize = ({ level, fontSize }) =>
  (fontSize ? getFontSize : `${(16 + (Math.floor((2 ** (6 - level)) / 2) * 2)) / 16}rem`)

const styles = css`
  font-family: ${fontFamily};
  font-weight: ${fontWeight};
  font-size: ${calculateFontSize};
  color: ${color};
  margin: 0;
`

const Heading = styled(({
  level, ...props
}) => React.createElement(`h${level}`, props))`${styles}`

Heading.propTypes = {
  level: PropTypes.number,
  font: PropTypes.string,
  fontSize: PropTypes.number,
  fontWeight: PropTypes.number,
}

Heading.defaultProps = {
  level: 1,
  font: 'myeongjo',
  fontSize: undefined,
  fontWeight: 600,
}

export default Heading
