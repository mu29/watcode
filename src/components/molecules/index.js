export { default as Field } from './Field'
export { default as IconButton } from './IconButton'
export { default as Tab } from './Tab'
