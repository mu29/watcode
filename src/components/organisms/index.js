export { default as ArtworkList } from './ArtworkList'
export { default as Header } from './Header'
export { default as LoginForm } from './LoginForm'
export { default as ReduxField } from './ReduxField'
export { default as RegisterForm } from './RegisterForm'
export { default as SearchBar } from './SearchBar'
