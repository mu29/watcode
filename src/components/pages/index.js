export { default as HomePage } from './HomePage'
export { default as RankingPage } from './RankingPage'
export { default as BookmarkPage } from './BookmarkPage'
