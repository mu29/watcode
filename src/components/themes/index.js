export { default as basicTheme } from './basic'
