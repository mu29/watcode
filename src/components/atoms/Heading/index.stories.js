import React from 'react'
import { storiesOf } from '@storybook/react'
import Heading from '.'

storiesOf('Atoms/Heading', module)
  .add('기본', () => (
    <Heading>작은 성공</Heading>
  ))
  .add('레벨 2', () => (
    <Heading level={ 2 }>작은 성공</Heading>
  ))
  .add('레벨 3', () => (
    <Heading level={ 3 }>작은 성공</Heading>
  ))
  .add('레벨 1, 2rem', () => (
    <Heading fontSize={ 32 }>작은 성공</Heading>
  ))
  .add('레벨 1, 얇게', () => (
    <Heading fontWeight={ 100 }>작은 성공</Heading>
  ))
