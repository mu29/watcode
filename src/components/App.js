import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { injectGlobal, ThemeProvider } from 'styled-components'
import { authorizeActions, readBookmarksActions } from 'store/actions'
import { auth } from 'services/firebase'
import { basicTheme } from './themes'
import Router from './Router'

/* eslint-disable no-unused-expressions */
injectGlobal`
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0;
  }
`

export default class App extends React.Component {
  static propTypes = {
    store: PropTypes.shape({
      dispatch: PropTypes.func,
      getState: PropTypes.func,
    }).isRequired,
  }

  componentWillMount() {
    const { store } = this.props
    auth.onAuthStateChanged((user) => {
      store.dispatch(authorizeActions.request({ user }))
      store.dispatch(readBookmarksActions.request())
    })
  }

  render() {
    const state = this.props.store.getState()
    const isLoggedIn = state.auth.user.email !== undefined
    return (
      <React.Fragment>
        <Helmet titleTemplate="watcode">
          <title>왓코드 - 오늘 볼 만화 추천해줘!</title>
          <meta charSet="UTF-8" />
          <meta name="description" content="" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <meta property="og:site_name" content="watcode" />
        </Helmet>
        <ThemeProvider theme={ basicTheme }>
          <Router isLoggedIn={ isLoggedIn } />
        </ThemeProvider>
      </React.Fragment>
    )
  }
}
