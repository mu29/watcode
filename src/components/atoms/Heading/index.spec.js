import React from 'react'
import { shallow } from 'enzyme'
import { basicTheme } from 'components/themes'
import Heading from '.'

const wrap = (props = {}) => shallow(<Heading theme={ basicTheme } { ...props } />).dive()

describe('<Heading />', () => {
  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = wrap({ children: 'test' })
    expect(wrapper.contains('test')).toBe(true)
  })

  it('임의의 속성이 들어온 경우 그려준다', () => {
    const wrapper = wrap({ name: 'foo' })
    expect(wrapper.find({ name: 'foo' })).toHaveLength(1)
  })

  it('h1을 기본으로 그려준다', () => {
    const wrapper = wrap()
    expect(wrapper.find('h1')).toHaveLength(1)
  })

  it('level 속성이 들어오면 hLevel 엘리먼트를 그려준다', () => {
    const wrapper = wrap({ level: 4 })
    expect(wrapper.find('h4')).toHaveLength(1)
  })

  it('fontSize 속성이 들어오면 해당 크기로 그려준다', () => {
    const wrapper = wrap({ fontSize: 16 })
    expect(wrapper.find({ fontSize: 16 })).toHaveLength(1)
  })
})
