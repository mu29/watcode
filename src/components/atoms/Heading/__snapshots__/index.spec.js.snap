// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> fontSize 속성이 들어오면 해당 크기로 그려준다 1`] = `
.c0 {
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-weight: 600;
  font-size: 1rem;
  margin: 0;
}

<h1
  className="c0"
  font="default"
  fontSize={16}
  fontWeight={600}
  theme={
    Object {
      "fonts": Object {
        "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
        "monospace": "'Monaco', 'Consolas'",
      },
      "palette": Object {
        "gray": Object {
          "0": "#FAFAFA",
          "10": "#F5F5F5",
          "100": "#212121",
          "20": "#EEEEEE",
          "30": "#E0E0E0",
          "40": "#BDBDBD",
          "60": "#9E9E9E",
          "70": "#757575",
          "80": "#616161",
          "90": "#424242",
        },
        "red": Object {
          "default": "#FF8787",
        },
        "transparent": Object {
          "default": "transparent",
        },
        "white": Object {
          "default": "#FFF",
        },
        "yellow": Object {
          "default": "#F9D14C",
        },
      },
      "sizes": Object {},
    }
  }
/>
`;

exports[`<Heading /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
.c0 {
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-weight: 600;
  font-size: 3rem;
  margin: 0;
}

<h1
  className="c0"
  font="default"
  fontWeight={600}
  theme={
    Object {
      "fonts": Object {
        "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
        "monospace": "'Monaco', 'Consolas'",
      },
      "palette": Object {
        "gray": Object {
          "0": "#FAFAFA",
          "10": "#F5F5F5",
          "100": "#212121",
          "20": "#EEEEEE",
          "30": "#E0E0E0",
          "40": "#BDBDBD",
          "60": "#9E9E9E",
          "70": "#757575",
          "80": "#616161",
          "90": "#424242",
        },
        "red": Object {
          "default": "#FF8787",
        },
        "transparent": Object {
          "default": "transparent",
        },
        "white": Object {
          "default": "#FFF",
        },
        "yellow": Object {
          "default": "#F9D14C",
        },
      },
      "sizes": Object {},
    }
  }
>
  test
</h1>
`;

exports[`<Heading /> 정상적으로 렌더링된다 1`] = `
.c0 {
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-weight: 600;
  font-size: 3rem;
  margin: 0;
}

<h1
  className="c0"
  font="default"
  fontWeight={600}
  theme={
    Object {
      "fonts": Object {
        "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
        "monospace": "'Monaco', 'Consolas'",
      },
      "palette": Object {
        "gray": Object {
          "0": "#FAFAFA",
          "10": "#F5F5F5",
          "100": "#212121",
          "20": "#EEEEEE",
          "30": "#E0E0E0",
          "40": "#BDBDBD",
          "60": "#9E9E9E",
          "70": "#757575",
          "80": "#616161",
          "90": "#424242",
        },
        "red": Object {
          "default": "#FF8787",
        },
        "transparent": Object {
          "default": "transparent",
        },
        "white": Object {
          "default": "#FFF",
        },
        "yellow": Object {
          "default": "#F9D14C",
        },
      },
      "sizes": Object {},
    }
  }
/>
`;
